virttest/utils_sriov.py

```diff
@@ -76,11 +76,11 @@ def add_or_del_connection(params, session=None, is_del=False):
     if not all([bridge_name, pf_name]):
         return
 
-    if not utils_package.package_install(["tmux"], session):
-        logging.error("Failed to install required package - tmux!")
+    if not utils_package.package_install(["tmux", "dhcp-client"], session):
+        logging.error("Failed to install the required package")
```

Comment:
The logging output is too generic,
Can we change to 
""Failed to install the required package:tumx or dhcp-client" ?

Reply:
@chunfuwen No need. Plenty of log could be found there.

Reply:
Yes, it is. There will be a lot of information, so I think it's unnecessary. @chunfuwen  please re-check it, thanks!

Reply:
> Yes, it is. There will be a lot of information, so I think it's unnecessary. @chunfuwen please re-check it, thanks!

@Yingshun
Could you share more specifics how can know "tmux", or "dhcp-client" install fail except logging.error explicitly expressed?

Reply:
The simplest way is to try:
```
L0604 INFO | Running 'rpm -q tmux'
2021-05-31 02:58:10,502 process          L0416 DEBUG| [stdout] package tmux is not installed
2021-05-31 02:58:10,503 process          L0685 INFO | Command 'rpm -q tmux' finished with 1 after 0.010106086730957031s
2021-05-31 02:58:10,504 process          L0604 INFO | Running '/usr/bin/dnf -y install tmux'
2021-05-31 02:58:10,617 process          L0416 DEBUG| [stderr] Error: There are no enabled repositories in "/etc/yum.repos.d", "/etc/yum/repos.d", "/etc/distro.repos.d".
2021-05-31 02:58:10,632 process          L0685 INFO | Command '/usr/bin/dnf -y install tmux' finished with 1 after 0.1231074333190918s
2021-05-31 02:58:10,633 utils_package    L0176 ERROR| Operate tmux on host failed
2021-05-31 02:58:11,634 process          L0604 INFO | Running 'rpm -q tmux'
2021-05-31 02:58:11,653 process          L0416 DEBUG| [stdout] package tmux is not installed
2021-05-31 02:58:11,653 process          L0685 INFO | Command 'rpm -q tmux' finished with 1 after 0.010291814804077148s
2021-05-31 02:58:11,654 process          L0604 INFO | Running '/usr/bin/dnf -y install tmux'
2021-05-31 02:58:11,771 process          L0416 DEBUG| [stderr] Error: There are no enabled repositories in "/etc/yum.repos.d", "/etc/yum/repos.d", "/etc/distro.repos.d".
2021-05-31 02:58:11,786 process          L0685 INFO | Command '/usr/bin/dnf -y install tmux' finished with 1 after 0.1257338523864746s
2021-05-31 02:58:11,786 utils_package    L0176 ERROR| Operate tmux on host failed
2021-05-31 02:58:12,788 process          L0604 INFO | Running 'rpm -q tmux'
2021-05-31 02:58:12,807 process          L0416 DEBUG| [stdout] package tmux is not installed
2021-05-31 02:58:12,807 process          L0685 INFO | Command 'rpm -q tmux' finished with 1 after 0.010590076446533203s
2021-05-31 02:58:12,808 process          L0604 INFO | Running '/usr/bin/dnf -y install tmux'
2021-05-31 02:58:12,925 process          L0416 DEBUG| [stderr] Error: There are no enabled repositories in "/etc/yum.repos.d", "/etc/yum/repos.d", "/etc/distro.repos.d".
2021-05-31 02:58:12,939 process          L0685 INFO | Command '/usr/bin/dnf -y install tmux' finished with 1 after 0.12589454650878906s
2021-05-31 02:58:12,940 utils_package    L0176 ERROR| Operate tmux on host failed
2021-05-31 02:58:13,942 process          L0604 INFO | Running 'rpm -q tmux'
2021-05-31 02:58:13,962 process          L0416 DEBUG| [stdout] package tmux is not installed
2021-05-31 02:58:13,963 process          L0685 INFO | Command 'rpm -q tmux' finished with 1 after 0.012292861938476562s
2021-05-31 02:58:13,964 process          L0604 INFO | Running '/usr/bin/dnf -y install tmux'
2021-05-31 02:58:14,081 process          L0416 DEBUG| [stderr] Error: There are no enabled repositories in "/etc/yum.repos.d", "/etc/yum/repos.d", "/etc/distro.repos.d".
2021-05-31 02:58:14,095 process          L0685 INFO | Command '/usr/bin/dnf -y install tmux' finished with 1 after 0.1257624626159668s
2021-05-31 02:58:14,096 utils_package    L0176 ERROR| Operate tmux on host failed
2021-05-31 02:58:15,097 process          L0604 INFO | Running 'rpm -q tmux'
2021-05-31 02:58:15,116 process          L0416 DEBUG| [stdout] package tmux is not installed
2021-05-31 02:58:15,117 process          L0685 INFO | Command 'rpm -q tmux' finished with 1 after 0.01049661636352539s
2021-05-31 02:58:15,118 process          L0604 INFO | Running '/usr/bin/dnf -y install tmux'
2021-05-31 02:58:15,235 process          L0416 DEBUG| [stderr] Error: There are no enabled repositories in "/etc/yum.repos.d", "/etc/yum/repos.d", "/etc/distro.repos.d".
2021-05-31 02:58:15,249 process          L0685 INFO | Command '/usr/bin/dnf -y install tmux' finished with 1 after 0.12631607055664062s
2021-05-31 02:58:15,250 utils_package    L0176 ERROR| Operate tmux on host failed
2021-05-31 02:58:16,252 process          L0604 INFO | Running 'rpm -q tmux'
2021-05-31 02:58:16,270 process          L0416 DEBUG| [stdout] package tmux is not installed
2021-05-31 02:58:16,271 process          L0685 INFO | Command 'rpm -q tmux' finished with 1 after 0.01124119758605957s
2021-05-31 02:58:16,272 process          L0604 INFO | Running '/usr/bin/dnf -y install tmux'
2021-05-31 02:58:16,389 process          L0416 DEBUG| [stderr] Error: There are no enabled repositories in "/etc/yum.repos.d", "/etc/yum/repos.d", "/etc/distro.repos.d".
2021-05-31 02:58:16,404 process          L0685 INFO | Command '/usr/bin/dnf -y install tmux' finished with 1 after 0.12720775604248047s
2021-05-31 02:58:16,404 utils_package    L0176 ERROR| Operate tmux on host failed
2021-05-31 02:58:17,406 process          L0604 INFO | Running 'rpm -q tmux'
2021-05-31 02:58:17,425 process          L0416 DEBUG| [stdout] package tmux is not installed
2021-05-31 02:58:17,426 process          L0685 INFO | Command 'rpm -q tmux' finished with 1 after 0.01092839241027832s
2021-05-31 02:58:17,427 process          L0604 INFO | Running '/usr/bin/dnf -y install tmux'
2021-05-31 02:58:17,544 process          L0416 DEBUG|
```
You will see a lot of error message like above.

Reply:
@Yingshun 
I notice that if those packages install failed, here it just print logging, do we need throw exception here instead of logging error since those packages are preconditions for further actions?

Reply:
Thank you for asking me this! At the begging of the design, we thought about this problem.  Here, what we are most concerned about is whether the shell command is success.  If the packages' installation fails and there are no tmux or dhclient binaries, the shell command can notice people with an proper message.

Reply:
@Yingshun 
As you mentioned previously that a lot of error message can notice people whether package install succeed or not.
If just logging error message, it means that it lost trackback where the issue firstly occurred.
Anyway, this is optional change, I can approve this PR even under such condition.